clippy_lints/src/formatting.rs

```diff
@@ -127,6 +149,28 @@ fn check_else_if(cx: &EarlyContext, expr: &ast::Expr) {
     }
 }
 
+/// Implementation of the `POSSIBLE_MISSING_COMMA_IN_ARRAY_FORMATTING` lint for array
+fn check_array(cx: &EarlyContext, expr: &ast::Expr) {
+    if let ast::ExprKind::Array(ref array) = expr.node {
+        for element in array {
+            if let ast::ExprKind::Binary(ref op, ref lhs, _) = element.node {
+                let space_span = mk_sp(lhs.span.hi, op.span.lo);
```

Comment:
You need to use `differing_macro_contexts`, otherwise you might end up creating a broken span if macros are used.